node/network/src/peer_manager.rs

```diff
@@ -435,4 +474,162 @@ mod tests {
             10000,
         );
     }
+
+    #[test]
```

Comment:
Nice test! Maybe later when we add more tests to check various conditions for banning we can create some helpers to make tests like this smaller.